rotkehlchen/chain/ethereum/decoding/decoder.py

```diff
@@ -485,40 +486,61 @@ def _maybe_decode_erc20_721_transfer(
                 action_items.pop(idx)
                 break  # found an action item and acted on it
 
+        # Add additional information to transfers for different protocols
+        self._maybe_enrich_transfers(
+            token=token,
+            tx_log=tx_log,
+            transaction=transaction,
+            event=transfer,
+            action_items=action_items,
+        )
         return transfer
 
     def _maybe_enrich_transfers(  # pylint: disable=no-self-use
             self,
-            token: Optional[EthereumToken],  # pylint: disable=unused-argument
+            token: Optional[EthereumToken],
             tx_log: EthereumTxReceiptLog,
-            transaction: EthereumTransaction,  # pylint: disable=unused-argument
-            decoded_events: List[HistoryBaseEntry],
-            action_items: List[ActionItem],  # pylint: disable=unused-argument
+            transaction: EthereumTransaction,
+            event: HistoryBaseEntry,
+            action_items: List[ActionItem],
     ) -> Optional[HistoryBaseEntry]:
         if tx_log.topics[0] == GTC_CLAIM and tx_log.address == '0xDE3e5a990bCE7fC60a6f017e7c4a95fc4939299E':  # noqa: E501
```

Comment:
This won't work. `_maybe_decode_erc20_721_transfer()`

starts with 

```python
        if tx_log.topics[0] != ERC20_OR_ERC721_TRANSFER:
            return None
```

That means that when you get in here the tx_log you have a topic[0] is a transfer.

The way the function is written it won't work due to the if checks checking `topic[0]` for something completely different.

The function needs to be split in 2.

One should be the function it was before and called where it was before.
And the other (your addition) needs to be another function, similar name, called as you have it now.

I am not sure if there is a way to do it better than what I am suggesting here. But as I said what you have coded does not work. Which also shows we need unit tests for all of these cases.

Reply:
Ah true, didn't notice that